src/Plugin/resource/Field/ResourceFieldBase.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal|restful\Plugin\resource\Field\ResourceFieldBase.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field;
+
+use Drupal\restful\Exception\ServerConfigurationException;
+use Drupal\restful\Http\Request;
+
+abstract class ResourceFieldBase implements ResourceFieldInterface {
```

Comment:
Actually, there is `ResourceFieldEntity` for that. In this class we only deal with the basics (all that applies to all data providers). `'property'` is supposed to act as a mapping to the _field_ identifier:

  - Field API field name. Entity provider.
  - Column name. DB provider.
  - Array key. Plugin provider.